buildroot/bin/env_clean

```diff
@@ -1,13 +1,13 @@
 #!/usr/bin/env bash
 
-[ -d ".pioenvs" ] && rm -r .pioenvs
-[ -d ".piolibdeps" ] && rm -r .piolibdeps
-[ -d ".piolib" ] && rm -r .piolib
+[ -d .pioenvs ] && rm -r .pioenvs
+[ -d .piolibdeps ] && rm -r .piolibdeps
+[ -d .piolib ] && rm -r .piolib
 
 if [ ! -z "$1" ]; then
   if [ $1 = "--deep" ]; then
-    [ -d "~/.platformio/packages" ] && rm -r ~/.platformio/packages/*
-    [ -d "~/.platformio/platforms" ] && rm -r ~/.platformio/platforms/*
-    [ -d "~/.platformio/.cache" ] && rm -r ~/.platformio/.cache/*
+    [ -d ~/.platformio/packages ] && [ ! -z "$(ls -A ~/.platformio/packages)" ] && rm -r ~/.platformio/packages/*
+    [ -d ~/.platformio/platforms ] && [ ! -z "$(ls -A ~/.platformio/platforms)" ] && rm -r ~/.platformio/platforms/*
+    [ -d ~/.platformio/.cache ] && [ ! -z "$(ls -A ~/.platformio/.cache)" ] && rm -r ~/.platformio/.cache/*
```

Comment:
If you include the `-f` flag in the `rm` command it won't return an error or interrupt the script. So this would be cleaner:
```sh
    rm -rf ~/.platformio/packages/*
    rm -rf ~/.platformio/platforms/*
    rm -rf ~/.platformio/.cache/*
```

Reply:
That is true of course, I just dislike having force in scripts as a rule when its not required